### PR TITLE
Convert test from karma to jest

### PR DESCRIPTION
**Pull request by fcollonval**
Closes #571

### CODE REVIEW

**Comment by fcollonval** [comment]
This is now ready for review JS tests are passing

**Review by vidartf — APPROVED** [review]
Thanks for this. I really like the karma-typescript setup when it is working work, but it has just been flaky for too long, so I think this is probably for the best.

I added two small comments that can be dealt with in a follow-up PR.

File: packages/nbdime/package.json [hunk 1]
```diff
@@ -37,23 +34,20 @@
     "json-stable-stringify": "^1.0.1"
   },
   "devDependencies": {
+    "@babel/core": "^7.5.0",
+    "@babel/preset-env": "^7.5.0",
+    "@jupyterlab/testutils": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (fcollonval) [reply packages/nbdime/package.json:39]: Indeed that was a mistake when copy-pasting dep from another project.

**Comment by fcollonval** [comment]
I opened #580 to address your review comments (thanks for them).

**Pull request merged**